store/sqlstore/upgrade.go

```diff
@@ -350,6 +351,17 @@ func UpgradeDatabaseToVersion47(sqlStore SqlStore) {
 		sqlStore.AlterColumnTypeIfExists("OAuthAuthData", "State", "varchar(1024)", "varchar(1024)")
 		sqlStore.RemoveColumnIfExists("ChannelMemberHistory", "Email")
 		sqlStore.RemoveColumnIfExists("ChannelMemberHistory", "Username")
+
+		defaultTimezone := make(map[string]string)
+		defaultTimezone["useAutomaticTimezone"] = "true"
+		defaultTimezone["automaticTimezone"] = ""
+		defaultTimezone["manualTimezone"] = ""
+
+		defaultTimezoneValue, err := json.Marshal(defaultTimezone)
+		if err != nil {
+			l4g.Critical(err)
+		}
+		sqlStore.CreateColumnIfNotExists("Users", "Timezone", "varchar(150)", "varchar(150)", string(defaultTimezoneValue))
```

Comment:
I think you'll also need to set the column size limit here https://github.com/mattermost/mattermost-server/blob/master/store/sqlstore/user_store.go#L81

Is 150 arbitrary or is that close to the maximum length of the field using the longest timezone, etc.?

Reply:
@jwilander, 150 refers to the closest maximum length with some padding (maybe like 6 char additional padding)

Could you tell me what's the difference between `varchar(150)` and `table.ColMap("Position").SetMaxSize(128)`

Reply:
>@jwilander, 150 refers to the closest maximum length with some padding (maybe like 6 char additional padding)

Any objections to bumping it to 256 just to be safe? It's a varchar so that space won't be taken up if it's not used

>Could you tell me what's the difference between varchar(150) and table.ColMap("Position").SetMaxSize(128)

Essentially there is no difference, but it needs to get added at the line I linked as well since that bit of code is for new installs. If we don't specify there it will default to something else